adr/0002-generate-extension-metadata-in-deployment-module.adoc

```diff
@@ -0,0 +1,123 @@
+= Enhanced extension metadata generation
+
+* Status: proposed
+* Date: 2024-04-26
+
+== Context and Problem Statement
+
+How do we know which extensions provide a dev service? At the moment, we don't, except by folk knowledge and https://quarkus.io/guides/dev-services[manually maintained lists.]
```

Comment:
this feels to me like just one minor out of bigger issues we have.

I feel like this is only taking into consideration what is in quarkus-extension.yaml where the actual overall issue is that we never really clearly defined which phases creates/generates what files and where they are stored. 

see https://github.com/quarkusio/quarkus/issues/4539 for (probably outodate and now even more tidbits of files) that are in play.

I'm not saying all of these files are bad or good - but given we are also doing work where config metadata is being treated differently i think its worth it we look at this more holistically than just the 'need to know if extension has dev service and runtime vs deployment generation challenges'

Reply:
I agree there are larger discussions that we could be having, but I don't think they need to happen all at once. I'd prefer to limit the scope of this ADR to "let's move the metadata generation to a stage where more information is available".